### PR TITLE
790 refactor errors: comment_operation

### PR DESCRIPTION
**Pull request by s-medvedev**

### CODE REVIEW

File: tests/tests/operation_tests.cpp [hunk 1]
```diff
@@ -452,29 +508,29 @@ BOOST_FIXTURE_TEST_SUITE(operation_tests, clean_database_fixture)
...
+            BOOST_CHECK_EQUAL( to_string( alice_content.body ), op.body );
+            //BOOST_CHECK_EQUAL( alice_content.json_metadata, op.json_metadata );
 #else
-            BOOST_REQUIRE(to_string(alice_content.title) == "");
-            BOOST_REQUIRE(to_string(alice_content.body) == "");
-            //BOOST_REQUIRE( alice_content.json_metadata == "" );
+            BOOST_CHECK_EQUAL(to_string(alice_content.title), "");
+            BOOST_CHECK_EQUAL(to_string(alice_content.body), "");
+            //BOOST_CHECK_EQUAL( alice_content.json_metadata, "" );
 #endif
```
Comment (afalaleev) [inline tests/tests/operation_tests.cpp]: Forgot on checking review from Anton. 
IS_LOW_MEM doesn't work anymore, please remove part with `#else  ... #endif`

File: libraries/chain/steem_evaluator.cpp [hunk 2]
```diff
@@ -563,19 +565,25 @@ namespace golos { namespace chain {
...
                 if (itr == by_permlink_idx.end()) {
                     if (o.parent_author != STEEMIT_ROOT_POST_PARENT) {
-                        FC_ASSERT(_db.get(parent->root_comment).allow_replies, "The parent comment has disabled replies.");
+                        GOLOS_CHECK_LOGIC(_db.get(parent->root_comment).allow_replies, 
+                                logic_exception::replies_are_not_allowed,
+                                "The parent comment has disabled replies.");
+
                         if (_db.has_hardfork(STEEMIT_HARDFORK_0_12__177) && !_db.has_hardfork( STEEMIT_HARDFORK_0_18__536) ) {
-                            FC_ASSERT(
+                            GOLOS_CHECK_LOGIC(
```
Comment (zxcat) [inline libraries/chain/steem_evaluator.cpp]: No chance to get here: all current operations have no restriction and all pre-HF18 ops are already valid. @afalaleev, maybe we should remove this `if` completely?

File: libraries/chain/steem_evaluator.cpp [hunk 3]
```diff
@@ -713,11 +724,13 @@ namespace golos { namespace chain {
                     const auto &comment = *itr;
                     if ( !_db.has_hardfork( STEEMIT_HARDFORK_0_18__536 ) ) {
                         if (_db.has_hardfork(STEEMIT_HARDFORK_0_14__306)) {
-                            FC_ASSERT(_db.calculate_discussion_payout_time(comment) != fc::time_point_sec::maximum(),
-                                      "The comment is archived.");
+                            GOLOS_CHECK_LOGIC(_db.calculate_discussion_payout_time(comment) != fc::time_point_sec::maximum(),
```
Comment (zxcat) [inline libraries/chain/steem_evaluator.cpp]: Will never get here too (whole `if ( !_db.has_hardfork( STEEMIT_HARDFORK_0_18__536 ) ) {…}`)

File: libraries/chain/steem_evaluator.cpp (hunk 2 above)
Comment (afalaleev) [reply libraries/chain/steem_evaluator.cpp]: Yes, you're right, this check can be removed.

File: libraries/chain/steem_evaluator.cpp (hunk 3 above)
Comment (afalaleev) [reply libraries/chain/steem_evaluator.cpp]: I agree, this check can be removed too.

**Pull request merged**